### PR TITLE
Add support for snat, dnat and ipmc crm resources

### PR DESCRIPTION
**Pull request by PrabhuSreenivasan**
Signed-off-by: Prabhu Sreenivasan <prabhu.sreenivasan@broadcom>

- What I did
Added support for snat, dnat and ipmc resources under CRM module.

- How I did it
New feature NAT adds new resources snat_enty and dnat_entry that needs to be monitored. ipmc_entry tracks IP multicast resources used by switch.

- How to verify it
sonic-utilities tests and crm spytest

- Associated PRs
Azure/sonic-swss#1511
Azure/sonic-utilities/pull/1258

### CODE REVIEW

File: files/build_templates/init_cfg.json.j2 [hunk 1]
```diff
@@ -10,7 +10,7 @@
             "polling_interval": "300",
 {%- for crm_res in ["ipv4_route", "ipv6_route", "ipv4_nexthop", "ipv6_nexthop", "ipv4_neighbor",
                     "ipv6_neighbor", "nexthop_group_member", "nexthop_group", "acl_table",
-                    "acl_group", "acl_entry", "acl_counter", "fdb_entry"] %}
+                    "acl_group", "acl_entry", "acl_counter", "fdb_entry", "snat_entry", "dnat_entry", "ipmc_entry"] %}
```
Comment (lguohan) [inline files/build_templates/init_cfg.json.j2:13]: what if snat_entry, dnat, ipmc_entry not enabled in the image? will there be any issue to have crm threshold there?

Comment (PrabhuSreenivasan) [reply files/build_templates/init_cfg.json.j2:13]: Orch will ignore these new entries on the config_db.json with a log message like the one given below.

Nov 24 15:56:48.149727 sonic ERR swss#orchagent: :- handleSetCommand: Failed to parse CRM Config configuration. Unknown attribute snat_entry_low_threshold.

**Comment by ben-gale** [comment]
All, @arlakshm - this has been quiet for 6 days - can we get some review input so we can move forwards please? The clock is ticking ...

**Comment by ben-gale** [comment]
All, @arlakshm - this has been quiet for 6 days - can we get some review input so we can move forwards please? The clock is ticking ...

**Comment by ben-gale** [comment]
@arlakshm , @lguohan, @rlhui  - What's the next step here? This has been inactive for 13 days now - how do we move towards merge pls?

**Review by arlakshm — APPROVED** [review]
LGTM

**Pull request merged**